virotour/lib/src/tour/tour_list_view.dart

```diff
@@ -1,70 +1,134 @@
+import 'dart:convert';
+
 import 'package:flutter/material.dart';
+import 'package:http/http.dart' as http;
 import 'package:virotour/src/settings/settings_view.dart';
 import 'package:virotour/src/tour/tour.dart';
 import 'package:virotour/src/tour/tour_details_view.dart';
 
-class TourListView extends StatelessWidget {
+class TourListView extends StatefulWidget {
   const TourListView({
-    super.key,
-    // TODO: Get the list of tours from GET /
-    this.items = const [Tour(1), Tour(2), Tour(3)],
-  });
+    Key? key,
+    required this.items,
+  }) : super(key: key);
 
   static const routeName = '/';
 
   final List<Tour> items;
 
+  @override
+  _TourListViewState createState() => _TourListViewState();
+}
+
+class _TourListViewState extends State<TourListView> {
+  late Future<List<Tour>> _tourData;
+
+  @override
+  void initState() {
+    super.initState();
+    _tourData = fetchData();
+  }
+
+  static Future<List<Tour>> fetchData() async {
+    // TODO: Implement fetching the tour data from an API endpoint
+    // final response = await http.get(Uri.parse('https://virotour.com/tours/endpoint'));
+    final response = [
+      {
+        'data': {
+          "tours": [
+            {
+              "tour_name": "example tour 1",
+              "description": "This is an example tour",
+              "id": 1
+            },
+            {
+              "tour_name": "example tour 2",
+              "description": "This is another example tour",
+              "id": 2
+            }
+          ]
+        }
+      }
+    ];
+
+    // if (response.statusCode == 200) {
+    //   final responseData = jsonDecode(response.body);
+    final data = response[0]['data']!['tours'] as List<dynamic>;
```

Comment:
Issue: Don't cast a nullable value to a non nullable type.

Reply:
This is only here as a placeholder until the API gets added.

Reply:
That's ok. Can you specify that with a comment and remove the unused code?

Reply:
Will do!